### PR TITLE
Fix a couple of life-time issues when plugin loading fails.

### PR DESCRIPTION
**Pull request by rsmmr**
Reported by Coverity.

Follow-up to #1179.

### CODE REVIEW

File: src/plugin/Manager.cc [hunk 1]
```diff
@@ -223,17 +227,17 @@ bool Manager::ActivateDynamicPluginInternal(const std::string& name, bool ok_if_
...
-			current_dir = nullptr;
-			current_sopath = nullptr;
 			current_plugin = nullptr;
-
 			DBG_LOG(DBG_PLUGINS, "  Loaded %s", path);
 			}
 
 		globfree(&gl);
+
+		if ( errors->size() )
```
Comment (timwoj) [inline src/plugin/Manager.cc:239]:
```suggestion
		if ( ! errors->empty() )
```

**Review by timwoj — CHANGES_REQUESTED** [review]
Did you mean to pull in changes to submodules in this PR?

**Comment by rsmmr** [comment]
> Did you mean to pull in changes to submodules in this PR?

Oops, no. Will fix.

File: src/plugin/Manager.cc (hunk 1 above)
Comment (rsmmr) [reply src/plugin/Manager.cc:239]: I seem to be the only person who prefers `.size()` over `! empty()`. :-) Will change.

**Pull request merged**